keras_retinanet/preprocessing/open_images.py

```diff
@@ -0,0 +1,206 @@
+                description_table[row[0]] = row[1].replace("\"", "").replace("'", "").replace('`', '')
+
+    with open(trainable_classes_path, 'rb') as f:
+        trainable_classes = f.read().split('\n')
+
+    id2labels_dict = dict([(i, description_table[c]) for i, c in enumerate(trainable_classes)])
+    cls_index_dict = dict([(c, i) for i, c in enumerate(trainable_classes)])
+
+    return id2labels_dict, cls_index_dict
+
+
+def generate_images_annotations_json(main_dir, metadata_dir, subset, cls_index_dict):
+    annotations_path = os.path.join(metadata_dir, subset, 'annotations-human-bbox.csv')
+
+    cnt = 0
+    with open(annotations_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile,
+                                fieldnames=['ImageID', 'Source', 'LabelName',
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+        for _ in reader:
+            cnt += 1
+
+        print ('There are {} lines in subset: {}'.format(cnt, subset))
+
+    id_annotations = dict()
+
+    with open(annotations_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile,
+                                fieldnames=['ImageID', 'Source', 'LabelName',
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+
+        images_sizes = dict()
+        for line, row in enumerate(reader):
+            frame = row['ImageID']
+
+            class_name = row['LabelName']
+
+            if class_name not in cls_index_dict:
+                continue
+
+            cls_id = cls_index_dict[class_name]
+
+            img_path = os.path.join(main_dir, 'images', subset, frame + '.jpg')
+            if frame in images_sizes:
+                width, height = images_sizes[frame]
+            else:
+                try:
+                    with Image.open(img_path) as img:
+                        width, height = img.width, img.height
+                        images_sizes[frame] = (width, height)
+                except Exception:
+                    continue
+
+            x1 = float(row['XMin'])
+            x2 = float(row['XMax'])
+            y1 = float(row['YMin'])
+            y2 = float(row['YMax'])
+
+            x1_abs = int(round(x1 * width))
+            x2_abs = int(round(x2 * width))
+            y1_abs = int(round(y1 * height))
+            y2_abs = int(round(y2 * height))
+
+            # Check that the bounding box is valid.
+            if x2 <= x1:
+                raise ValueError('line {}: x2 ({}) must be higher than x1 ({})'.format(line, x2, x1))
+            if y2 <= y1:
+                raise ValueError('line {}: y2 ({}) must be higher than y1 ({})'.format(line, y2, y1))
+
+            if y2_abs == y1_abs:
+                print ('filtering line {}: rounding y2 ({}) and y1 ({}) makes them equal'.format(line, y2, y1))
```

Comment:
I prefer to print warnings with `warnings`, like [here](https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/preprocessing/generator.py#L101).